### PR TITLE
Bumping the toolchain version 1.76

### PR DESCRIPTION
**Pull request by batconjurer**
## Describe your changes
Moving to rust version 1.76.0
## Indicate on which release or other PRs this topic is based on
0.31.6

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Comment by Fraccaman** [comment]
hvae u seen https://github.com/anoma/namada/pull/2566 ?

**Comment by batconjurer** [comment]
> hvae u seen #2566 ?

Nope totally missed that. I needed these changes for something unrelated, so I decided to bang em out. This already has all the clippy fixes, I just need to check the tests locally

File: crates/core/src/types/token.rs [hunk 1]
```diff
@@ -584,12 +590,12 @@ impl PartialOrd for DenominatedAmount {
...
                 if rem.is_zero() {
-                    Some(Ordering::Equal)
+                    Ordering::Equal
                 } else {
-                    Some(Ordering::Greater)
+                    Ordering::Greater
                 }
             } else {
-                ord
+                ord.unwrap()
```
Comment (tzemanovic) [inline crates/core/src/types/token.rs:598]: this doesn't seem right - `ord.is_none()`, no?

Comment (tzemanovic) [reply crates/core/src/types/token.rs:598]: could be*

**Comment by Fraccaman** [comment]
we need to split docker changes into a separete PR and merge those before updating the toolchain

File: crates/core/src/types/token.rs (hunk 1 above)
Comment (tzemanovic) [reply crates/core/src/types/token.rs:598]: ah, the partial ord on `Uint` is always `Some`, but we can just use `cmp` in place of `partial_cmp`

Comment (tzemanovic) [reply crates/core/src/types/token.rs]: done in [2a6095f](https://github.com/anoma/namada/pull/2687/commits/2a6095f06865254e0707210ff10e032031e1ad08)

Comment (batconjurer) [reply crates/core/src/types/token.rs]: there was a new lint about `cmp` that I had to fix a few times. I just kinda ran through it, so they may not always be the most elegant.

File: crates/macros/src/lib.rs [hunk 2]
```diff
@@ -389,7 +389,7 @@ mod test_proc_macros {
         let expected_impl = quote! {
             impl Keys {
                 /// A list of all storage keys
-                pub const ALL: &[&'static str] = {
+                pub const ALL: &'static [&'static str] = {
```
Comment (sug0) [inline crates/macros/src/lib.rs:392]: careful with those safe rust transmutations :DD

**Comment by tzemanovic** [comment]
> we need to split docker changes into a separete PR and merge those before updating the toolchain

done in https://github.com/anoma/namada/pull/2696

**Review by tzemanovic — APPROVED** [review]

**Review by sug0 — COMMENTED** [review]
some nits

File: crates/core/src/types/key/secp256k1.rs [hunk 3]
```diff
@@ -490,16 +490,16 @@ impl Hash for Signature {
...
     }
 }
 
 impl Ord for Signature {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.partial_cmp(other).unwrap()
+        match self.0.to_bytes().partial_cmp(&other.0.to_bytes()) {
+            Some(Ordering::Equal) => self.1.cmp(&other.1),
+            res => res.unwrap(),
+        }
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply crates/core/src/types/key/secp256k1.rs:502]: This isn't correct. If self.0 and other.0 are equal, we need to compare self.1 with other.1

[the remark this answers is not in the document]
Comment (sug0) [reply crates/core/src/types/key/secp256k1.rs:502]: ah, I didn't see the other comparison, sry. hm but either way we can still avoid calling `partial_cmp` I think, it just wasn't caught by clippy

Comment (batconjurer) [inline crates/core/src/types/key/secp256k1.rs:502]:
```suggestion
        match self.0.to_bytes().cmp(&other.0.to_bytes()) {
            Ordering::Equal => self.1.cmp(&other.1),
            res => res,
        }
```

**Comment by tzemanovic** [comment]
I squashed the fixup commit and rebased onto 0.31.6

**Pull request merged**